lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -510,13 +510,11 @@ withNetworkLayer tr np addrInfo (versionData, _) action = do
             :: Show e
             => String
             -> IO (Either e r)
-            -> ExceptT ErrNetworkUnavailable IO r
+            -> ExceptT ErrStakeDistribution IO r
         handleQueryResult label =
             withExceptT mkErr . ExceptT . bracketQuery label tr
           where
-            -- fixme: "network unreachable" is not the best name for this. And
-            -- the exception message is ignored by the API server.
-            mkErr e = ErrNetworkUnreachable $ T.pack $ "Unexpected " ++ show e
+            mkErr = ErrStakeDistributionQuery . T.pack . show
```

Comment:
👌 I think this would always be `AcquireFailure` though

Reply:
Thank you very much for pointing this out @Anviking.
I believe that you have identified the source of flakiness.
I have created a ticket ADP-647 to fix it.